packages/concerto-core/lib/decoratormanager.js

```diff
@@ -296,12 +318,36 @@ class DecoratorManager {
      */
     static executeCommand(namespace, declaration, command) {
         const { target, decorator, type } = command;
-        const { name } = ModelUtil.parseNamespace(namespace);
-        if (
-            this.falsyOrEqual(target.namespace, [namespace, name]) &&
-            this.falsyOrEqual(target.declaration, [declaration.name])
-        ) {
-            if (!target.property && !target.type) {
+        const { name } = ModelUtil.parseNamespace( namespace );
+        if (this.falsyOrEqual(target.namespace, [namespace,name]) &&
+            this.falsyOrEqual(target.declaration, [declaration.name])) {
+
+            if (declaration.$class === 'concerto.metamodel@1.0.0.MapDeclaration') {
```

Comment:
const { MetaModelNamespace } = require('@accordproject/concerto-metamodel');

Can we please use `${MetaModelNamespace}.MapDeclaration`?

Reply:
I was looking for that, thanks!